### PR TITLE
Use LabelDataset naming convention

### PR DESCRIPTION
**Pull request by yuyu2172**
Follows discussion in #399.

ClassificationDataset has been added after updating the version to `0.6.0`.
Therefore, this PR is *not* `no-compat`.

### CODE REVIEW

File: docs/source/reference/utils.rst [hunk 1]
```diff
@@ -79,6 +75,10 @@ assert_is_image
 ~~~~~~~~~~~~~~~
 .. autofunction:: assert_is_image
 
+assert_is_label_dataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Hakuyume) [inline docs/source/reference/utils.rst]: The length of `~`

Comment (yuyu2172) [reply docs/source/reference/utils.rst]: Thank you!

File: docs/source/reference/datasets.rst [hunk 2]
```diff
@@ -3,9 +3,9 @@ Datasets
 
 .. module:: chainercv.datasets
 
-DirectoryParsingClassificationDataset
+DirectoryParsingLabelDataset
 -------------------------------------
```
Comment (Hakuyume) [inline docs/source/reference/datasets.rst]: Sorry. This line should also be fixed. (the length of `-`)

Comment (yuyu2172) [reply docs/source/reference/datasets.rst]: Oh. Sorry...

File: chainercv/datasets/directory_parsing_label_dataset.py [hunk 3]
```diff
@@ -91,8 +91,8 @@ class DirectoryParsingClassificationDataset(chainer.dataset.DatasetMixin):
             --- class_1
                 |-- img_0.png
 
-        >>> from chainercv.dataset import DirectoryParsingClassificationDataset
-        >>> dataset = DirectoryParsingClassificationDataset('root')
+        >>> from chainercv.dataset import DirectoryParsingLabelDataset
```
Comment (Hakuyume) [inline chainercv/datasets/directory_parsing_label_dataset.py]: `from chainercv.dataset` -> `from chainercv.datasets`?

**Comment by yuyu2172** [comment]
@Hakuyume 
Can you merge?

**Pull request merged**